### PR TITLE
[DNM] MATERIAL DATUMS

### PR DESCRIPTION
**Pull request by Iamgoofball**
MEMES

qaejqkm

### CODE REVIEW

**Comment by Iamgoofball** [comment]
okay guys

anyone got any ideas for crap we can add for the Big One?

**Comment by duncathan** [comment]
revert goonchem :^)

**Comment by optimumtact** [comment]
YOU DON'T MAKE THE MEME PULL WITHOUT HAVING AN IDEA GOOF

**Comment by Iamgoofball** [comment]
@optimumtact but my idea was to secure the slot so that you couldn't get it and we could actually use it for something good

**Comment by Anonus** [comment]
revert goonmemes

**Comment by optimumtact** [comment]
okay lets at least come up with something good

**Comment by duncathan** [comment]
better idea: revert goonchem, port baymed :^)

**Comment by Ikarrus** [comment]
Reported.

**Comment by Aranclanos** [comment]
what the hell dude you said you were preparing something like three months ago this isn't even funny christmas is ruined

**Comment by Fox-McCloud** [comment]
A reworked, functional, and mechanically interesting Thermo Electric Generator.

**Comment by ike709** [comment]
Fixes #9968

**Comment by ike709** [comment]
My amazing ideas for #10000:
* Port baymed
* Finish literally any of RR's unfinished features (Preferably Comp 3.3 or dismemberment)
* Add more dank memes
* Remove existing dank memes
* Automatic Thunderdome fighting tournament system for ghosts that you can watch on the station
More to come as I think of them... Also I'm pretty sure someone else suggested or attempted to implement that last one at some point. I can't remember.

**Comment by Iamgoofball** [comment]
i got an idea

**Comment by ike709** [comment]
What's your idea?

**Comment by Iamgoofball** [comment]
you'll see

**Comment by Niknakflak** [comment]
Is it tossing you off your high horse that you think a number in github makes you cool because your shitty code says otherwise?
No?
Ok
 revert retardchem

**Comment by Niknakflak** [comment]
Revert (~~Retard~~)(~~Autistic~~)(~~terrible~~)(~~Goof~~)GoonChem

**Comment by PKPenguin321** [comment]
make this PR just straight up remove every line of code in the game

number 10,000, the ultimate fun removal

**Comment by Iamgoofball** [comment]
No, I've got an actually serious thing planned. I'll be committing it soonish.

**Comment by Jordie0608** [comment]
You're a massive dork for making all those issues to pad up the issue count.

**Comment by Iamgoofball** [comment]
#9990 and #9991 are valid, reopen those @Jordie0608

**Comment by ike709** [comment]
Is #9998 valid too?

**Comment by ike709** [comment]
what do material datums do

**Comment by Fox-McCloud** [comment]
It's basically like Bay and TG's materials systems where everything is made of a particular type of material and certain materials have certain strengths (or special properties).

Ie: gold adds a sparkly overlay to the item (or whoever is holding it), silver grants disease resistance---other things give high force ratings, etc.

**Comment by ike709** [comment]
That's cool but I don't know if it is really #10000 worthy

**Comment by Iamgoofball** [comment]
@ike709 end goal is literally everything uses the material system

Walls, floors, windows, clothing, armor, objects, everything 

Bonus goal is alloys and infusing reagents

**Comment by Fox-McCloud** [comment]
@Iamgoofball 

Oh no..not reagentshit.

**Comment by Iamgoofball** [comment]
@Fox-McCloud nono, think goon arc smelter infusing rragents

**Comment by Fox-McCloud** [comment]
@Iamgoofball 

yeah, that's what the goon community/nerds specifically called 'reagentshit'.

**Comment by ike709** [comment]
That's more like it Goof. Good job.

**Comment by Dorsisdwarf** [comment]
Ahahaahaha
this is what you made a bunch of fake issues to put in?
That's hilarious. Couldn't handle the thought of oranges beating your shitty PR, right?

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 1]
```diff
@@ -204,8 +217,10 @@ var/global/list/datum/stack_recipe/silver_recipes = list ( \
 	throw_range = 3
 	origin_tech = "materials=4"
 	sheettype = "clown"
+	material = new/datum/material/bananium()
 
 var/global/list/datum/stack_recipe/clown_recipes = list ( \
+	new/datum/stack_recipe("toolbox", /obj/item/weapon/storage/toolbox/material, 1), \
 	new/datum/stack_recipe("bananium tile", /obj/item/stack/tile/mineral/bananium, 1, 4, 20), \
 	new/datum/stack_recipe("Clown Statue", /obj/structure/statue/bananium/clown, 5, one_per_turf = 1, on_floor = 1), \
 	)
```
Comment (MrPerson) [inline code/game/objects/items/stacks/sheets/mineral.dm]: I really hope I don't need to explain why this system is unfeasible long-term

File: code/game/objects/items/weapons/storage/toolbox.dm [hunk 2]
```diff
@@ -17,6 +17,32 @@
...
+	material = new/datum/material/diamond()
+/obj/item/weapon/storage/toolbox/material/plasma
+	material = new/datum/material/plasma()
+/obj/item/weapon/storage/toolbox/material/uranium
+	material = new/datum/material/uranium()
+/obj/item/weapon/storage/toolbox/material/bananium
+	material = new/datum/material/bananium()
+/obj/item/weapon/storage/toolbox/material/glass
+	material = new/datum/material/glass()
+
```
Comment (MrPerson) [inline code/game/objects/items/weapons/storage/toolbox.dm]: What the fuck, just set the material var after construction. These should not be separate types.

Comment (Iamgoofball) [inline code/game/objects/items/weapons/storage/toolbox.dm]: These are debug things I had in there so I didn't have to spawn and varedit a stack over and over again, it does assign the variable in construction

File: code/game/objects/items/stacks/sheets/mineral.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/game/objects/items/stacks/sheets/mineral.dm]: Oh yeah I hate this part of stack code

File: code/modules/materials/materials.dm [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+					I.flags |= CONDUCT
+					I.siemens_coefficient = material.conduction
+		material.my_atom = src
+	return
+
+/atom/Destroy()
+	if(material)
+		if(material.can_process)
+			SSobj.processing -= src
+	..()
```
Comment (MrPerson) [inline code/modules/materials/materials.dm]: You're forgetting to also qdel the material and null the var.

File: README.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
 [![Build Status](https://travis-ci.org/tgstation/-tg-station.png)](https://travis-ci.org/tgstation/-tg-station)
 
 
-**Website:** http://www.tgstation13.org <BR>
+**Website:** http://wiki.baystation12.net/Tajaran_Culture <BR>
```
Comment (RemieRichards) [inline README.md]: No

**Comment by Miauw62** [comment]
oh no goofball is doing what i planned on doing so now it will inevitably be shit.

**Comment by Ricotez** [comment]
fuck and I was working so hard to get Dismemberment ready for #10000

and instead we get this shit

:-1:

**Comment by Razharas** [comment]
What is the purpose of this system?

**Comment by Midaychi** [comment]
Goofball adding fun with materials so you can build diamond dildos, :-1:

**Comment by Firecage** [comment]
@Iamgoofball Just remember. If this gets closed for some reason, I am going to murder you with a clown horn for ruining #10000

**Comment by vista-** [comment]
>glass shattering

Finally, I can throw glassware at the goddamn clown properly. Make it so glass shards have a high chance of embedding in the victim :^)

File: README.md (hunk 4 above)
Comment (ike709) [inline README.md]: That's the one change he made so he could have a PR up to have #10000. He didn't actually have an idea, and he forgot to revert this placeholder.

File: code/modules/materials/materials.dm [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+		P.Insert(O, iconstate) //Build a new icon set to use with the item itself.
+
+	A.icon = P //Apply the new icon.
+	return
+
+/datum/material/diamond
+	name = "Diamond"
+	color = "#BFFFFF"
+	force_multiplier = 2
+	health_multiplier = 4
```
Comment (Anonus) [inline code/modules/materials/materials.dm]: Diamonds are relatively fragile, this should be at most 1.

File: code/modules/materials/materials.dm [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+	color = "#E122D9"
+	force_multiplier = 1
+	health_multiplier = 1
+	prefix = "plasma"
+
+/datum/material/silver
+	name = "Silver"
+	color = "#DCDCDC"
+	force_multiplier = 1
+	health_multiplier = 1
```
Comment (Anonus) [inline code/modules/materials/materials.dm]: Silver has similar properties to gold and should have similar stats.

File: code/modules/materials/materials.dm [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+	color = "#E7C88F"
+	force_multiplier = 1
+	health_multiplier = 1
+	prefix = "sandstone"
+
+/datum/material/uranium
+	name = "Uranium"
+	color = "#39C639"
+	force_multiplier = 3
+	health_multiplier = 5
```
Comment (Anonus) [inline code/modules/materials/materials.dm]: Both should probably be a bit lower, this ain't minecraft buddy

File: code/modules/materials/materials.dm (hunk 5 above)
Comment (CorruptComputer) [inline code/modules/materials/materials.dm]: diamonds != fragile

Comment (Anonus) [inline code/modules/materials/materials.dm]: Compared to glass diamonds aren't fragile, but they're definitely not unbreakable wizard gems; they can and do break if hit right, even if you're hitting them with something less hard.

**Comment by WJohn** [comment]
I can foresee literally a hundred bugs coming from this where shit deletes and there's massive runtimes and crashes EVERYWHERE OH GOD.

File: code/modules/materials/materials.dm (hunk 7 above)
Comment (Iamgoofball) [inline code/modules/materials/materials.dm]: Depleted uranium is dense

Comment (Miauw62) [inline code/modules/materials/materials.dm]: so is gold

Comment (Iamgoofball) [inline code/modules/materials/materials.dm]: I dunno, coderbus said uranium was good for weapons and crap.

These values are highly unlikely to be finalized values.

Comment (Firecage) [inline code/modules/materials/materials.dm]: Miauw, Gold is fragile though.

Comment (Midaychi) [inline code/modules/materials/materials.dm]: Well, fragile in the sense that it bends easily. Like pure iron, but softer.

**Comment by Memendia** [comment]
https://tgstation13.org/phpBB/viewtopic.php?f=5&t=3843&p=96363#p96363

;-;  Nooo Goofball why didn't you add the dildos!!

File: code/modules/materials/materials.dm (hunk 7 above)
Comment (optimumtact) [inline code/modules/materials/materials.dm]: This isn't depleted uranium though

Comment (Razharas) [inline code/modules/materials/materials.dm]: Natural uranium is exactly as hard but also a tiny bit lighter than depleted one

File: code/modules/materials/materials.dm (hunk 5 above)
Comment (pudl) [inline code/modules/materials/materials.dm]: throwing a diamond at the ground with any kind of force will shatter it

**Comment by Niknakflak** [comment]
revert goofchem

File: code/modules/materials/materials.dm (hunk 5 above)
Comment (duncathan) [inline code/modules/materials/materials.dm]: Depends on the shape, but for the most part cut diamonds will

**Comment by RemieRichards** [comment]
I've been speaking with goof, unable to really get my points across, but I feel the larger aspects of this PR are needless and provide no worthwhile changes to the game.

The clean up of copypaste and a mineral strings is all fine and great, I just don't feel that diamond/gold/uranium/silver chairs windows grilles rods tables etc. add anything to the game except needless complexity.

I'm only saying this here because goof wanted my pov known.

**Comment by Aranclanos** [comment]
you can also make static datums instead of having each object create a new one, that would avoid the if(material) on each proc that you want a specific material action

**Comment by Miauw62** [comment]
[kek intensifies]

**Comment by Razharas** [comment]
Ah miauw you cheat

**Comment by Aranclanos** [comment]
githuuuuuuuuuuuuuub

**Comment by JJRcop** [comment]
It's @Dorsisdwarf's fault!
There should be codebase rules for abusing github to close things you don't like!

**Comment by RemieRichards** [comment]
since when in the fuck did a #fixes ever be able to close a PR? hahaha.

**Comment by Thunder12345** [comment]
It's done it for as long as I've used git, but yeah, it's a problem when people abuse it to do this.

**Comment by Miauw62** [comment]
I swear I tried this once and it didnt work.

**Comment by Iamgoofball** [comment]
Hey what the fuck man

That was a dick move

**Comment by RemieRichards** [comment]
Dorsis should have a talking too, because that's not on.

**Comment by Niknakflak** [comment]
This should be closed to spite goofball because he cheated with the issues.  Punish him and have him remake the PR

File: code/modules/materials/materials.dm [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+/datum/material/gold
+	name = "Gold"
+	color = "" // Handled by the material init. proc.
+	force_multiplier = 0.25
+	health_multiplier = 0.25 // this aint digmineconstructcraft buddy
+	prefix = "gold"
+	conduction = 1
+	default_stack = /obj/item/stack/sheet/mineral/gold
+
```
Comment (GunHog) [inline code/modules/materials/materials.dm]: This should have the "unacidable" var, if you were yoinking the code from my EUK PR. It used gold to make things acid-proof.

File: code/modules/materials/materials.dm [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+	health_multiplier = 1
+	prefix = "silver"
+	default_stack = /obj/item/stack/sheet/mineral/silver
+/datum/material/bananium
+	name = "Bananium"
+	color = "#FFD800"
+	force_multiplier = 1
+	health_multiplier = 1
+	prefix = "bananium"
+	default_stack = /obj/item/stack/sheet/mineral/bananium
```
Comment (GunHog) [inline code/modules/materials/materials.dm]: Everything made of Bananium should have a clown- related sound!

File: code/modules/materials/materials.dm [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+	active_tick++
+	return
+
+/datum/material/plasma
+	name = "Plasma"
+	color = "#E122D9"
+	force_multiplier = 1
+	health_multiplier = 1
+	prefix = "plasma"
+	default_stack = /obj/item/stack/sheet/mineral/plasma
```
Comment (GunHog) [inline code/modules/materials/materials.dm]: You made this flammable, right? Right?

Comment (Thunder12345) [inline code/modules/materials/materials.dm]: Flaming plasma swords when

Comment (pudl) [inline code/modules/materials/materials.dm]: Isn't it already established that the only plasma-related thing that's flammable is the gas form? The dust and ore are both harmless.

Comment (GunHog) [inline code/modules/materials/materials.dm]: Things made from *solid* plasma are flammable. The liquid is not. The ore is not. The ingot is not, but structures (tiles, statues, walls, airlocks, doors) are all flammable.

**Comment by Dorsisdwarf** [comment]
Oh my god that's hilarious that git tracks "fixes XXX" in the commit names too.

but @mad people, that commit predates this PR. I used my ninja powers to predict this would be shitty, clearly.

File: code/modules/materials/materials.dm (hunk 10 above)
Comment (Dorsisdwarf) [inline code/modules/materials/materials.dm]: Pretty sure liquid plasma is flammable, like welding fluid and ethanol

**Comment by JJRcop** [comment]
@Dorsisdwarf Oh, alright. I'm not mad anymore because you didn't do it intentionally.

**Comment by Miauw62** [comment]
im not mad because its funny

**Comment by demonfiren** [comment]
0/10 no dragon dildos.

**Comment by Memendia** [comment]
But Dildos are the future my friends!

**Comment by ike709** [comment]
is #10000 ded

**Comment by Midaychi** [comment]
You could just port Bay dev's version

**Comment by Aranclanos** [comment]
Well this PR adds a bunch of if(material) snowflake checks in a bunch of files so its really unacceptable

**Comment by Iamgoofball** [comment]
i no longer have the will to trudge through shitcode considering even if this is finished it won't be merged

**Comment by WJohn** [comment]
You ruined the grand 10,000 goofball ;~;

**Comment by Iamgoofball** [comment]
it was ruined anyways

**Comment by Miauw62** [comment]
no, you ruined it.
you specifically made a bunch of bullshit issues so you could get this, and you ruined it.

**Comment by JJRcop** [comment]
I actually was prepared to get this, but I missed it when I went to a party somewhere. 
#### https://github.com/JJRcop/-tg-station/commit/e334830cd0099b28dc59abf74d97aed53f23e326

**Comment by ike709** [comment]
maybe some day

**Comment by outbools** [comment]
nevar

**Comment by optimumtact** [comment]
sack of wine

**Comment by Fikou** [comment]
cheese keyzza

**Pull request closed without merging**